src/app/reference-data/reference-data.component.ts

```diff
@@ -13,18 +13,40 @@ import { Observable } from 'rxjs';
 export class ReferenceDataComponent {
   referenceDataList$: Observable<RefrenceDataResponseDto>;
   referenceDataList: ReferenceDataModel[];
+  selectedRefDataType: string;
 
-  constructor(private modalService:NgbModal, private apiService: RealestateApiService){}
+constructor(private modalService:NgbModal, private apiService: RealestateApiService){}
   
-  ngOnInit(): void {
+ngOnInit(): void {
   this.apiService.getAllReferenceData().subscribe((referenceDataList$: any) => {
```

Comment:
referenceDataList$ is not an observable.

you should rename it to "response" for instance.

Reply:
Done.